pipelines/translation.py

```diff
@@ -0,0 +1,262 @@
+from kfp import components, dsl
+from kubernetes.client import V1Toleration
+from kubernetes.client.models import V1PersistentVolumeClaimVolumeSource
+
+from . import training
+from .common import *
+
+
+@dsl.pipeline(name='Full multilingual pipeline for dialogue datasets', description='Generate, Translate, Train, and Eval')
+def generate_translate_train_eval_pipeline(
+    owner='',
+    project='',
+    experiment='',
+    s3_bucket='geniehai',
+    source='user',
+    input_splits='train eval',
+    model_name_or_path='Helsinki-NLP/opus-mt-$(src_lang)-$(tgt_lang)',
```

Comment:
Let's call this translation model? Also, why do I have to specify both "marian" and the actual model name? Can we guess one from the other?

Reply:
Oh so it's a user chosen ID. Why is this not keyed off the normal `model` field?

Reply:
That's what the "dataset" parameter is then. Like, if it is used just to make S3 directories to write into, let's make it "${dataset}-nmt-${tgt_lang}" and be done with it. One fewer parameter to type.

Reply:
Agree on "translation_model". I just don't understand why you keep the extra parameter for the dataset ID there though?

Reply:
It's servers a similar purpose as "model" for training except this is for translation. Honestly, don't worry about it. I can give it a default value if you don't wanna be bothered by what to pass here every time.